### ISSUE TITLE
LibWeb: Canvas draws transparent images with a white background

### ISSUE THREAD

**Issue by jamierocks** [LadybirdBrowser/ladybird#1104]
| Ladybird | Expected (Firefox) |
|----|----|
| ![image](https://github.com/user-attachments/assets/d6610679-057a-4b8e-8a47-5284ddcf2c5b) | ![image](https://github.com/user-attachments/assets/6114addf-a644-4cd0-ae8d-e24a55fb05d3) |

## Minimal reproduction

```html
<script type="module">
    const canvas = document.querySelector('#canvas');
    const ctx = canvas.getContext('2d');

    const img = new Image();
    img.addEventListener("load", () => {
        ctx.drawImage(img, 10, 10, 100, 151);
    });
    img.src = 'https://housepoints-v2.pages.dev/assets/bar-chart/Apollo.png';
</script>
<style>
    html, body {
        height: 100%;
    }
    #canvas {
        background-image: url('https://housepoints-v2.pages.dev/assets/config/imperial-wallpaper.jpg');
        width: 100%;
        height: 100%;
    }
</style>

<canvas id="canvas"></canvas>
```

**Comment by tcl3** [comment]
This appears to be a regression. 

Bisected to: 5865cf5

Cc: @gmta 

For reference, prior to this commit, the above image looked like this in Ladybird:
![image](https://github.com/user-attachments/assets/e39d1ac7-ed45-4d4b-9cf0-84d2fb9be67e)

**Comment by gmta** [comment]
Whoops, will need to check! @awesomekling has a skia branch that we need to compare this with as well.

**Comment by awesomekling** [comment]
FWIW here it is on my Skia-backed canvas branch:
<img width="1030" alt="image" src="https://github.com/user-attachments/assets/8f4eb565-2414-44cb-9d76-9fd411610d17">

Looks like some premul/unpremul tweaks still needed :)

**Comment by circl-lastname** [comment]
Mostly fixed by #1024

**Comment by gmta** [comment]
@circl-lastname I've got a final fix incoming for this issue.
